dpctl/CMakeLists.txt

```diff
@@ -125,6 +125,8 @@ function(build_dpctl_ext _trgt _src _dest)
             )
         endif()
     endif()
+    # needed to prevent warnings from Cython caused by cimporting from Numpy
+    target_compile_definitions(${_trgt} PRIVATE NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION)
```

Comment:
I think a better approach would to include `npy_no_deprecated_api` header before `cimport numpy`, see

https://github.com/IntelPython/mkl_random/blob/8bc5e7c6b121df883e77ad53e02e943906922c7d/mkl_random/mklrand.pyx#L39